src/main/java/org/jvnet/hudson/test/JenkinsRule.java

```diff
@@ -606,6 +606,21 @@ public File getWebAppRoot() throws Exception {
     }
 
     /**
+     * Prepares a {@link WebAppContext} to be used in thehosting environment to get the
+     * {@link javax.servlet.ServletContext} implementation that we need for testing.
+     */
+    protected WebAppContext createWebAppContext() throws Exception {
+        final WebAppContext context = new WebAppContext(WarExploder.getExplodedDir().getPath(), contextPath);
+        context.setClassLoader(getClass().getClassLoader());
```

Comment:
Generally speaking the `getClass().getClassLoader()` idiom is unsafe, since it could cause a subclass to behave differently merely because it was defined in a child `ClassLoader`, without even overriding anything at all.

Use `JenkinsRule.class.getClassLoader()` unless you really have a reason to look for the loader of a rule subclass specifically.